convert.go

```diff
@@ -0,0 +1,147 @@
+// Package flatfs is a Datastore implementation that stores all
+// objects in a two-level directory structure in the local file
+// system, regardless of the hierarchy of the keys.
+package flatfs
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+)
+
+func Convert(oldPath string, getOldDir ShardFunc, newPath string, getNewDir ShardFunc, out io.Writer) error {
+	old, err := New(oldPath, getOldDir, false)
+	if err != nil {
+		return err
+	}
+	new, err := New(newPath, getNewDir, false)
+	if err != nil {
+		return err
+	}
+
+	if out != nil {
+		fmt.Fprintf(out, "Getting Keys...\n")
+	}
+	res, err := old.Query(query.Query{KeysOnly: true})
+	if err != nil {
+		return err
+	}
+	entries, err := res.Rest()
```

Comment:
For example using a ipfs repo I created on https://github.com/ipfs/go-ipfs/issues/3462 the time to gather the keys was around 2.5 minutes and it looks like the rename is going to take around 20 minutes.